cpp/src/arrow/engine/substrait/serde_test.cc

```diff
@@ -185,7 +185,8 @@ TEST(Substrait, SupportedExtensionTypes) {
     ASSERT_OK_AND_ASSIGN(
         auto buf,
         internal::SubstraitFromJSON(
-            "Type", "{\"user_defined_type_reference\": " + std::to_string(anchor) + "}"));
+            "Type", "{\"user_defined\": { \"type_reference\": " + std::to_string(anchor) +
+                        ", \"nullability\": \"NULLABILITY_NULLABLE\" } }"));
```

Comment:
`nullability` did we missed including this last time? Or are we planning to expose this too? My knowledge is not quite adequate here, appreciate some comments.

Reply:
No, Substrait itself didn't consider user-defined types to conceptually have nullability, for no particular reason I can think of. See https://github.com/substrait-io/substrait/pull/217

Reply:
I see. Let’s go with what you have now.